### DISCUSSION TITLE
"global" call to collect all the sdks (`workspace/sdks`)

### DISCUSSION THREAD

**Discussion by abrams27** [build-server-protocol/build-server-protocol#634] [Proposals]
so it's something similar to https://github.com/build-server-protocol/build-server-protocol/discussions/632 - instead of providing the same sdk over and over for each build target using `data` field we could just create one "global" endpoint for it and build targets could depend on it (via `dependencies` field or we can create a new one `sdk`)

it also makes it easier to implement on the client side (at least in intellij) where we need to calculate unique sdks and then add them to the project

what do you think about it?

**Comment by lefou** [comment]
I guess each build tool has other properties and it's pretty common that not all support the same flexibility. But why should we restrict the protocol just because of that? If you have a global sdk, just return it for each target. It's up to the BSP client to detect, that all target specific sdks are the same and deduplicate it, if necessary.

But how should such an endpoint be implemented by a BSP server, that does not have global sdks? 
What would be the expected consistency constraint for target-specific sdks vs. global sdks?
IMHO, providing more than one way to retrieve the same thing makes the protocol more complicated to work with and I think we should not extend it with redundant requests as the proposed `workspace/sdks`.

**Reply by abrams27** [reply]
well here reasoning is similar to https://github.com/build-server-protocol/build-server-protocol/discussions/632 and the solution is similar as well - if it's needed it's still possible to create a 1-1 mapping between build target and sdk (e.g. `<build target name>-<skd name>`)

**Reply by agluszak** [reply]
>  if it's needed it's still possible to create a 1-1 mapping between build target and sdk (e.g. <build target name>-<skd name>)

I'm not sure if I'm a fan of this solution. It won't be coherent with other requests. If want to go further down this road, we'd need to write this down in the spec, otherwise it's going to cause a lot pain for implementers
